nemo/collections/llm/tools/auto_configurator/core/training_config.py

```diff
@@ -70,89 +76,120 @@ def generate_grid_search_configs(
+        logger.debug(f"Processing TP={tp}, PP={pp}, CP={cp}, EP={ep}")
+        valid_virtual_pipelines = []
+
+        if virtual_pipeline_parallel_sizes is not None:
+            for vp_size in virtual_pipeline_parallel_sizes:
+                if num_layers % (pp * vp_size) == 0:
+                    valid_virtual_pipelines.append(vp_size)
+        if not valid_virtual_pipelines:
+            (
+                calculated_virtual_pipelines,
+                act_ckpt_layers,
+                num_micro_batches_partial_act_ckpt,
+                act_ckpt_layers_per_pipeline,
+            ) = _set_activations_checkpoint_params(
+                tp,
+                pp,
+                cp,
+                ep,
+                num_layers,
+                act_method,
+                multiplier,
+                model_size_in_b,
+                model_name,
+            )
+            valid_virtual_pipelines = (
+                [calculated_virtual_pipelines] if calculated_virtual_pipelines is not None else [None]
+            )
+        else:
+            (
+                _,
+                act_ckpt_layers,
+                num_micro_batches_partial_act_ckpt,
+                act_ckpt_layers_per_pipeline,
+            ) = _set_activations_checkpoint_params(
+                tp,
+                pp,
+                cp,
+                ep,
+                num_layers,
+                act_method,
+                multiplier,
+                model_size_in_b,
+                model_name,
+            )
+        for virtual_pipelines in valid_virtual_pipelines:
+            for mbs in params.mbs:
+                for gbs in params.gbs:
+                    num_gpus = base_cfg.trainer.num_nodes * base_cfg.trainer.devices
+                    model_parallelism = (tp * pp * cp * ep) if (cp and ep) else (tp * pp)
+                    mod_gbs = gbs % (mbs * num_gpus / model_parallelism)
+                    if mod_gbs != 0:
+                        continue
+
+                    kwargs = {
+                        "base_cfg": base_cfg,
+                        "act": None,
+                        "num_mbs_act": None,
+                        "act_per_pipe": None,
+                        "tp": tp,
+                        "pp": pp,
+                        "cp": cp,
+                        "ep": ep,
+                        "virtual_pipelines": virtual_pipelines,
+                        "mbs": mbs,
+                        "gbs": gbs,
+                        "max_steps": max_steps,
+                        "num_nodes": num_nodes,
+                        "path_to_logs": path_to_logs,
+                        "model_name": model_name,
+                        "model_size": model_size_in_b,
+                    }
+
+                    act_layers = None
+                    if act_ckpt_layers[0] is not None:
+                        if act_layers is not None and act_layers != "auto":
```

Comment:
You define `act_layers = None` two lines above this - will this branch ever get triggered?